src/AppInstallerCLICore/Commands/PinCommand.cpp

```diff
@@ -107,11 +107,12 @@ namespace AppInstaller::CLI
 
     void PinAddCommand::ValidateArgumentsInternal(Execution::Args& execArgs) const
     {
+        Argument::ValidateCommonArguments(execArgs);
+
         if (execArgs.Contains(Execution::Args::Type::GatedVersion) && execArgs.Contains(Execution::Args::Type::BlockingPin))
         {
             throw CommandException(Resource::String::BothGatedVersionAndBlockingFlagProvided);

```

Comment:
I absolutely love what you did with the string builder for the exclusive arguments, it looks great and once this PR gets merged I can take a look at what other exclusive arguments need to be validated. 

I know that this specific line isn't part of the PR, but these scenarios could also be a good fit for validating exclusive arguments. I don't think it needs to be added in this PR, I can try and do it on my own, but I was wondering how you would recommend going about it. Would it be best to create a function overload that accepted the string ID as a third parameter, or would it be better to implement it as an optional defaulting to nullopt?